### PR TITLE
[FEATURE] Enables teacher to reset student password

### LINKED ISSUE #1736: [UI idea] Reset password for kid by teacher
**Idea**

Some kids forget their password, but they can't access their email address (f.e. because they can't receive links or can't receive email outside of school) If would be great if teachers can set a password reset for this kid so that they can log in again. Once logged in, they can change their password again.

### PR DESCRIPTION
**Pull request by TiBiBa**
**Description**
In this PR we implement a new feature, enabling teachers too reset a students password. As discussed earlier some children don't have access to a mailadres at the school, making it hard to invite them to classes and/or reset their passwords. With the implementation of #1764 the first one is fixed. In this PR we fix the other issue.

We implement an additional column on the class-overview page called `change_password` pressing it will pop-up the modal for the teacher to enter a new password. Due to the table getting crowded we also implement dynamic column hiding/showing as already implemented within the admin page in #1740.

**Fix for**
This PR fixes #1736.

**How to test**
Navigate to the class-overview and notice the new attribute selection too at the top of the page. Make sure the "change password" column is shown and verify that you can indeed change the password of a student.

### CODE REVIEW

**Comment by MarleenGilsing** [comment]
I can't open the class page, @TiBiBa. I can go to the teachers page, but the class page gives me an error.

**Comment by TiBiBa** [comment]
> I can't open the class page, @TiBiBa. I can go to the teachers page, but the class page gives me an error.

Weird! It all works fine for me. @Felienne can you verify if the class page is broken?

**Comment by MarleenGilsing** [comment]
It's probably something with my computer @TiBiBa because I can't open the page either when I'm on Main instead of your PR.

**Comment by Felienne** [comment]
> It's probably something with my computer @TiBiBa because I can't open the page either when I'm on Main instead of your PR.

I know this does not help in any way but for me it works fine locally. What is the error @MarleenGilsing?

**Comment by MarleenGilsing** [comment]
I get this: 

image
image

**Comment by TiBiBa** [comment]
> I get this: 
> 
> 
> 
> image
> 
> image
> 
> 

I think I know the cause! For some reason you don't have the BASE_URL set as an environment variable within your IDE. What IDE are you using? There should be an option to add this value (it should be there, but not sure why it's not found).

**Comment by Felienne** [comment]
Ah ja ik zie het al! Jij hebt de BASE_URL niet ingesteld. Ik denk dat ik je dat wel eens heb laten zien in PyCharm kan je ergens je run configuration aanpassen, daar heb je denk ik al ADMIN_USER staat en daar moet je ook een BASE_URL bij doen, die je dan op "localhost:8080" moet instellen. Als dat niet lukt laat ik het donderdag wel zien.

**Comment by Felienne** [comment]
> > I get this:
> > image
> > image
> 
> I think I know the cause! For some reason you don't have the BASE_URL set as an environment variable within your IDE. What IDE are you using? There should be an option to add this value (it should be there, but not sure why it's not found).

It was not needed before, so I never showed Marleen how to set it. 

Of course if would also be nice if we did not fail this hard for other new contributors, maybe 1) add the needed env vars (also ADMIN_USER) to the docs and 2) throw a nicer error?

**Comment by MarleenGilsing** [comment]
Gelukt :)

**Comment by TiBiBa** [comment]
> > > I get this:
> > > image
> > > image
> > 
> > 
> > I think I know the cause! For some reason you don't have the BASE_URL set as an environment variable within your IDE. What IDE are you using? There should be an option to add this value (it should be there, but not sure why it's not found).
> 
> It was not needed before, so I never showed Marleen how to set it.
> 
> Of course if would also be nice if we did not fail this hard for other new contributors, maybe 1) add the needed env vars (also ADMIN_USER) to the docs and 2) throw a nicer error?

Agree. Propose to change it into the following:

`os.getenv('BASE_URL', 'http://0.0.0.0:8080')`

In case it is not set it falls back to the development url

**Comment by MarleenGilsing** [comment]
Now that I'm on the page, I can easily add the 'change password' column. 
Weirdly in my first few tries the button didn't work, but now it works perfectly! 

Small remark: It asks if I'm REALLY sure of this choice, which sounds a bit dramatic, haha, like I can't change it back. Maybe remove 'really' and 'heel'?

File: coursedata/texts/en.yaml [hunk 1]
```diff
@@ -281,6 +281,11 @@ Auth:
     number_programs: "Number of programs"
     amount_programs: "Saved programs"
     latest_shared_program: "Latest shared program"
+    visible_columns: "Visible columns"
+    enter_password: "Enter a new password for"
+    password_change_prompt: "Are you really sure you want to change this password?"
```
Comment (Felienne) [inline coursedata/texts/en.yaml:286]:
```suggestion
    password_change_prompt: "Are you sure you want to change this password?"
```

File: coursedata/texts/nl.yaml [hunk 2]
```diff
@@ -273,6 +273,11 @@ Auth:
     highest_level_reached: "Hoogste level"
     number_programs: "Aantal programma's"
     latest_shared_program: "Laatst gedeelde"
+    visible_columns: "Zichtbare kolommen"
+    enter_password: "Vul een nieuw wachtwoord in voor"
+    password_change_prompt: "Weet je heel zeker dat je dit wachtwoord wilt wijzigen?"
```
Comment (Felienne) [inline coursedata/texts/nl.yaml:278]:
```suggestion
    password_change_prompt: "Weet je zeker dat je dit wachtwoord wilt wijzigen?"
```

**Review by Felienne — APPROVED** [review]

**Comment by Felienne** [comment]
> Now that I'm on the page, I can easily add the 'change password' column. Weirdly in my first few tries the button didn't work, but now it works perfectly!

This will typically be the case since your browser remembers (in computer terms: caches) the code so it can take a while for it to see the now code.

> Small remark: It asks if I'm REALLY sure of this choice, which sounds a bit dramatic, haha, like I can't change it back. Maybe remove 'really' and 'heel'?

Good point, removed!

**Pull request merged**